networking/p2p/src/integration-test/java/tech/pegasys/artemis/networking/p2p/jvmlibp2p/GossipMessageHandlerIntegrationTest.java

```diff
@@ -80,20 +77,31 @@ public void shouldGossipBlocksAcrossToIndirectlyConnectedPeers() throws Exceptio
     eventBus1.post(newBlock);
 
     // Listen for new block event to arrive on networks 2 and 3
-    final Collection<BeaconBlock> propagatedBlocks = new ConcurrentLinkedQueue<>();
+    final BeaconBlockCollector network2Blocks = new BeaconBlockCollector(eventBus2);
+    final BeaconBlockCollector network3Blocks = new BeaconBlockCollector(eventBus3);
+
+    // Verify the expected block was gossiped across the network
     Waiter.waitFor(
         () -> {
-          final ArgumentCaptor<BeaconBlock> blockCaptor =
-              ArgumentCaptor.forClass(BeaconBlock.class);
-          verify(eventBus2, atLeastOnce()).post(blockCaptor.capture());
-          verify(eventBus3, atLeastOnce()).post(blockCaptor.capture());
-          propagatedBlocks.addAll(blockCaptor.getAllValues());
```

Comment:
ArgumentCaptor doesn't do any type checking so this was capturing all events published to the EventBus and only worked previously because the only events being published were `BeaconBlock` events. The `StoreInitializedEvent` is now firing and causing the test to fail.  Rather than fight mockito, I've just introduced a simple listener that can register to the real eventBus.

Reply:
that's unexpected - good to know!